hyperspy/drawing/_widgets/polygon.py

```diff
@@ -0,0 +1,168 @@
+        disconnect from the events. If turning on, the widget will connect to
+        the default events.
+        """
+        if value is not self.is_on and self.ax is not None:
+            if value is True:
+                self.blit = (
+                    hasattr(self.ax, "hspy_fig") and self.ax.figure.canvas.supports_blit
+                )
+                self.connect(self.ax)
+                if render_figure:
+                    self.ax.figure.canvas.draw_idle()
+            elif value is False:
+                self.disconnect()
+                if render_figure:
+                    self.ax.figure.canvas.draw_idle()
+                self.ax = None
+        self._is_on = value
+
+    def set_mpl_ax(self, ax):
+        """
+        Parameters
+        ----------
+        mpl_ax: matplotlib.axes._subplots.AxesSubplot
+            The `matplotlib` axis that the `PolygonWidget` will attach to.
+        """
+        if ax is self.ax or ax is None:
+            return  # Do nothing
+        # Disconnect from previous axes if set
+        if self.ax is not None and self.is_on:
+            self.disconnect()
+        self.ax = ax
+
+        self.set_on(True, render_figure=False)
+
+        # Colors of widget. Usually set from constructor.
+        handle_props = dict(color=self._color)
+        line_props = dict(color=self._color)
+
+        self._widget = PolygonSelector(
+            ax,
+            onselect=self._onselect,
+            useblit=self.blit,
+            handle_props=handle_props,
+            props=line_props,
+        )
+
+        self.ax.figure.canvas.draw_idle()
+
+    def set_vertices(self, vertices):
+        """Function for deleting the currently saved polygon and setting a new one.
+
+        Parameters
+        ----------
+        vertices : list of tuples
+            List of  `(x, y)` tuples of the vertices of the polygon to add.
+        """
+
+        if self.ax is not None and self.is_on:
+            if len(vertices) > 2:
+
+                if Version(matplotlib.__version__) >= Version("3.6"):
+                    self._widget.verts = list(vertices)
+                else:
+                    # Accessing some private members to be able to change the `matplotlib`
+                    # widgets with older `matplotlib` versions.
+                    # Can hopefully be removed soon in the future
+                    self._widget._xs, self._widget._ys = [
+                        list(dimension) for dimension in zip(*vertices)
+                    ]
+
+                    # Close the polygon
+                    self._widget._xs.append(self._widget._xs[0])
+                    self._widget._ys.append(self._widget._ys[0])
+
+                    self._widget._selection_completed = True
+                    self._widget.set_visible(True)
+                    self._widget._draw_polygon()
+                self._onselect(vertices)
+            self.ax.figure.canvas.draw_idle()
+
+    def get_vertices(self):
```

Comment:
Wondering if this should be a property/setter called `vertices`, to have a similar interface to other ROIs, for example, `SpanROI.left`, etc. The idea would be to get/set the parameters of ROI in a similar by using the name of the parameter - here `vertices`.

Reply:
Yes, this is a good idea and should be implemented for consistency. So I have done just that!